source/jormungandr/jormungandr/street_network/geovelo.py

```diff
@@ -34,42 +34,42 @@
 import json
 from navitiacommon import response_pb2
 from jormungandr import app
-from jormungandr.exceptions import TechnicalError, InvalidArguments
+from jormungandr.exceptions import TechnicalError, InvalidArguments, UnableToParse
 from jormungandr.street_network.street_network import AbstractStreetNetworkService
-from jormungandr.utils import get_pt_object_coord, is_url, record_external_failure
+from jormungandr.utils import get_pt_object_coord, is_url
 
 
 class Geovelo(AbstractStreetNetworkService):
-    sn_system_id = 'geovelo'
 
-    def __init__(self, instance, service_url, timeout=10, api_key=None, **kwargs):
+    def __init__(self, instance, service_url, id='geovelo', timeout=10, api_key=None, **kwargs):
         self.instance = instance
+        self.sn_system_id = id
         if not is_url(service_url):
-            raise ValueError('service_url is invalid, you give {}'.format(service_url))
+            raise ValueError('service_url {} is not a valid url'.format(service_url))
         self.service_url = service_url
         self.api_key = api_key
         self.timeout = timeout
         self.breaker = pybreaker.CircuitBreaker(fail_max=app.config['CIRCUIT_BREAKER_MAX_GEOVELO_FAIL'],
                                                 reset_timeout=app.config['CIRCUIT_BREAKER_GEOVELO_TIMEOUT_S'])
 
-    def _pt_object_summary(self, pt_object):
+    @staticmethod
+    def _pt_object_summary(pt_object):
         coord = get_pt_object_coord(pt_object)
         return [coord.lat, coord.lon, getattr(pt_object, 'uri', None)]
 
-    def _make_data(self, origins, destinations):
-        data = {}
-        data['starts'] = []
-        for o in origins:
-            data['starts'].append(self._pt_object_summary(o))
-        data['ends'] = []
-        for d in destinations:
-            data['ends'].append(self._pt_object_summary(d))
-        return data
+    @staticmethod
+    def _make_request_arguments(origins, destinations):
+        return {
+            'starts': [Geovelo._pt_object_summary(o) for o in origins],
```

Comment:
this is valid, but you can also make it a classmethod and take `cls` as the first argument of the method